packages/catalog-generator/src/main/java/io/kaoto/camelcatalog/maven/KaotoMavenVersionManager.java

```diff
@@ -0,0 +1,177 @@
+            String gav = String.format("%s:%s:%s", groupId, artifactId, version);
+
+            boolean useSnapshots = version.endsWith("SNAPSHOT");
+            resolve(mavenDownloader, gav, useSnapshots, false);
+
+            if (artifactId.contains("catalog")) {
+                this.version = version;
+            } else {
+                this.runtimeProviderVersion = version;
+            }
+
+            return true;
+        } catch (Exception e) {
+            if (getLog()) {
+                LOGGER.log(Level.WARNING,
+                        String.format("Cannot load runtime provider version {} due {}", version, e.getMessage()), e);
+            }
+            return false;
+        }
+    }
+
+    /**
+     * Resolves Maven artifact using passed coordinates and use downloaded artifact
+     * as one of the URLs in the
+     * helperClassLoader, so further Catalog access may load resources from it.
+     */
+    public void resolve(MavenDownloader mavenDownloader, String gav, boolean useSnapshots, boolean transitive) {
+        try {
+            Set<String> extraRepositories = new LinkedHashSet<>(repositories.values());
+
+            List<MavenArtifact> artifacts = mavenDownloader.resolveArtifacts(Collections.singletonList(gav),
+                    extraRepositories, transitive, useSnapshots);
+
+            if (getLog()) {
+                LOGGER.log(Level.INFO, () -> "Artifacts: " + artifacts);
+            }
+
+            for (MavenArtifact ma : artifacts) {
+                ((KaotoOpenURLClassLoader) getClassLoader()).addURL(ma.getFile().toURI().toURL());
+            }
+        } catch (Throwable e) {
+            if (getLog()) {
+                LOGGER.log(Level.WARNING,
+                        String.format("Cannot resolve artifact {} due to {}", gav, e.getMessage()), e);
+            }
+        }
+
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String name) {
+        InputStream is = null;
+
+        if (runtimeProviderVersion != null) {
+            is = doGetResourceAsStream(name, runtimeProviderVersion);
+        }
+        if (is == null && version != null) {
+            is = doGetResourceAsStream(name, version);
+        }
+        if (getClassLoader() != null && is == null) {
+            is = getClassLoader().getResourceAsStream(name);
+        }
+
+        return is;
+    }
+
+    private InputStream doGetResourceAsStream(String name, String version) {
+        if (version == null) {
+            return null;
+        }
+
+        try {
+            URL found = null;
+            Enumeration<URL> urls = getClassLoader().getResources(name);
+            while (urls.hasMoreElements()) {
+                URL url = urls.nextElement();
+                if (url.getPath().contains(version)) {
+                    found = url;
+                    break;
+                }
+            }
+            if (found != null) {
+                return found.openStream();
+            }
```

Comment:
can directly return found.openStream instead of using a `found` and a `break`

Reply:
can you be more precise here?

Reply:
```
    private InputStream doGetResourceAsStream(String name, String version) {
        if (version != null) {
            try {
                Enumeration<URL> urls = getClassLoader().getResources(name);
                while (urls.hasMoreElements()) {
                    URL url = urls.nextElement();
                    if (url.getPath().contains(version)) {
                        return url.openStream();
                    }
                }
                
            } catch (IOException e) {
                if (getLog()) {
                    LOGGER.log(Level.WARNING, String.format("Cannot open resource {} and version {} due {}", name, version,
                            e.getMessage(), e));

                }
            }
        }
        return null;
```
?